offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/person/update/UpdatePersonView.groovy

```diff
@@ -32,6 +38,69 @@ class UpdatePersonView extends CreatePersonView{
     private void adjustViewElements() {
         submitButton.caption = "Update Person"
         abortButton.caption = "Abort Person Update"
+
+        //add a grid
+        affiliations = new Grid<>()
+        generateAffiliationGrid()
+        this.addComponent(affiliations,2)
+        //add the add button
+    }
+
+    private void generateAffiliationGrid() {
+        try {
+            this.affiliations.addColumn({ affiliation -> affiliation.organisation })
+                    .setCaption("Organisation").setId("Organisation")
+            this.affiliations.addColumn({ affiliation -> affiliation.addressAddition })
+                    .setCaption("Address Addition").setId("AddressAddition")
+            this.affiliations.addColumn({ affiliation -> affiliation.street })
+                    .setCaption("Street").setId("Street")
+            this.affiliations.addColumn({ affiliation -> affiliation.city })
+                    .setCaption("City").setId("City")
+            this.affiliations.addColumn({ affiliation -> affiliation.postalCode })
+                    .setCaption("Postal Code").setId("PostalCode")
+            this.affiliations.addColumn({ affiliation -> affiliation.country })
+                    .setCaption("Country").setId("Country")
+
+            //specify size of grid and layout
+            affiliations.setWidthFull()
+            affiliations.setHeightMode(HeightMode.ROW)
+            affiliations.setHeightByRows(5)
+
+        } catch (Exception e) {
+            new Exception("Unexpected exception in building the affiliation grid", e)
+        }
+
+        /*
+        Let's not forget to setup the grid's data provider
+        */
+        def affiliationDataProvider = setupAffiliationDataProvider()
+        /*
+        Lastly, we add some content filters for the columns
+         */
+        addFilters(affiliationDataProvider)
+    }
+
+    private ListDataProvider setupAffiliationDataProvider() {
+        def affiliationListDataProvider = new ListDataProvider<>(updatePersonViewModel.affiliationList)
+        this.affiliations.setDataProvider(affiliationListDataProvider)
+
+        return affiliationListDataProvider
+    }
+
+    private void addFilters(ListDataProvider affiliationListDataProvider) {
+        HeaderRow personFilterRow = affiliations.appendHeaderRow()
+        GridUtils.setupColumnFilter(affiliationListDataProvider,
+                affiliations.getColumn("Organisation"),
+                personFilterRow)
+        GridUtils.setupColumnFilter(affiliationListDataProvider,
+                affiliations.getColumn("AddressAddition"),
+                personFilterRow)
+        GridUtils.setupColumnFilter(affiliationListDataProvider,
+                affiliations.getColumn("Street"),
+                personFilterRow)
+        GridUtils.setupColumnFilter(affiliationListDataProvider,
+                affiliations.getColumn("City"),
+                personFilterRow)
```

Comment:
Is there a reason why the filters are not extended over the entire grid?

Reply:
No, it was just for testing in the beginning and I forgot to add the filter to all columns. Good catch